spec/api/v1/identities/get_identity_by_ldap_uid_spec.rb

```diff
@@ -0,0 +1,184 @@
+require 'spec_helper'
+
+RSpec.describe 'SPARCCWF::APIv1', type: :request do
+  
+  describe 'GET /v1/identities.json' do
+    before do
+      FactoryGirl.create_list(:identity, 5)
+    end
+    
+    context ':shallow record,' do
+       context 'valid ldap_uid' do
+        before do
+          @identity = Identity.first
+          params = { :depth => 'shallow',
+                     :limit => 1,
+                     :query => { ldap_uid: @identity.ldap_uid } 
+                   }
+           cwf_sends_api_get_request_for_resources_by_params('identities', params) 
+        end
+        it 'should respond with a single identity' do
+          parsed_body         = JSON.parse(response.body)
+          expected_attributes = ['callback_url', 'sparc_id']
+
+          expect(parsed_body['identity'].keys.sort).to eq(expected_attributes)
+          expect(parsed_body['identity']['sparc_id']).to eq(@identity.id)
+        end
+      end
+      
+      context 'empty ldap_uid' do
+       before do
+         @identity = Identity.first
+         params = { :depth => 'shallow',
+                    :limit => 1,
+                    :query => { ldap_uid: "" } 
+                  }
+          cwf_sends_api_get_request_for_resources_by_params('identities', params) 
+       end
+       it 'should respond with an error' do
+         parsed_body         = JSON.parse(response.body)
+         expect(parsed_body['identity']).to eq(nil)
+         expect(parsed_body['error']).to eq("Identity not found for query #<Hashie::Mash ldap_uid=\"\">")
```

Comment:
To provide query details to the client to help with debugging. For example, it will help when third party app end users report issues but are unable to provide adequate details; because this information will be in our Java log files. How does that sound?